pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -182,12 +193,12 @@ func (np *NetworkPolicy) ruleSelectsPeer(rulePeers []netv1.NetworkPolicyPeer, pe
 			} else {
 				selector, err := np.parseNetpolLabelSelector(rulePeers[i].PodSelector)
 				if err != nil {
-					return false, err
+					return false, false, err
 				}
 				peerMatchesPodSelector = selector.Matches(labels.Set(peer.GetPeerPod().Labels))
 			}
 			if peerMatchesPodSelector {
-				return true, nil //  matching both pod selector and ns_selector here
+				return true, anyNsMatch, nil //  matching both pod selector and ns_selector here
```

Comment:
if the function returns here on the first rule (which is not an "all ns" match), we can miss an "all ns" match if it is in the second rule?

Reply:
seemed to me, that always the (on first call) first peer sent to this function (to be checked if selected) is the `IP-block peer` - and whether it is selected or not, I continued to check for "all ns" with the `entireClusterCheckOnly` flag.

(from this assumption I didn't add the same check with `entireClusterCheckOnly` flag here too.); shouldn't I assume the IP-block will be always sent first?

Reply:
but if first rule is `from IPBlock` , even though the peer is IP-type; it will miss the `all ns` check (so at least on that return satement `line 217` I need to fix )

Reply:
and will not assume first peer is always IP (and fix for this rule too `line 201`)

Reply:
I think this function implementation might be having too many assumptions, causing this to be error prone.. I wouldn't rely on having a call with an IP peer, and definitely not on the order.

Reply:
I'd consider a separate function to check policy rules only once for this specific goal, on the first time we start analyzing this policy (store the results so we do not repeat those computations).

Reply:
not sure I understand (2) - can you reference to an example?

Reply:
suupose we have following netpol:

> 
```
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: allow-hello-world-b-to-a-app
  namespace: hello-world
spec:
  podSelector:
    matchLabels:
      app: a-app
  policyTypes:
  - Ingress
  - Egress
  ingress:
  - from:
    - namespaceSelector: {}
    ports:
    - port: 8050
      protocol: TCP
  egress:
  - to:
    - namespaceSelector: {}
```

and only one peer in the resources `deployment with app: a-app`.

the only peers we'll get for looping in the `connlist.getConnectionsBetweenPeers` are the `a-app deployment and full IP-block`.
since the netpol restricts the conns only in the cluster; the allowed connections computed between the a-peer an ip-peer
(`in connlist line 504: allowedConnections, err := pe.AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer))` 
will be empty; and hence we'll not get to the call in `connlist 518 : p2pConnection, err := ca.checkIfP2PConnOrExposureConn(pe, allowedConnections, srcPeer, dstPeer, exposuresMap)`. 
so I had to update the `a-app peer` entire cluster connections; also when allowed conns between "existing" peers are empty. 
(so I had to add the check in `connlist lines 508-513 calling updatePeersGeneralExposureData(pe, srcPeer, dstPeer, ingressSet, egressSet, exposuresMap)`

Reply:
ok, so both (1) and (2) are related to the same fix in `connlist`?

Reply:
yes;
because of 2 ; i had to move the call (`updatePeersGeneralExposureData`) from `checkIfP2PConnOrExposureConn` to here.
and because of 1; i had to update also "unprotected" data in this call (not only entireClusterConns)